docs/source/user/stencil.rst

```diff
@@ -0,0 +1,205 @@
+``a[x+0,y+0]`` in the input array.  Similarly, ``a[-1,1]`` in the stencil
+kernel corresponds to ``a[x-1,y+1]`` in the input array.
+
+Depending on the specified kernel, the kernel cannot be applied to the
+borders of the output image as this may cause the input array to be
+accessed out-of-bounds.  How the stencil decorator handles this situation
+is dependent upon which :ref:`stencil-mode` is selected.  
+The default mode is for the stencil decorator to set the border elements 
+of the output array to zero.
+
+To invoke a stencil on an input array, call the stencil as if it were
+a regular function and pass the input array as the argument::
+
+   >>> import numpy as np
+   >>> input_arr = np.arange(100).reshape((10, 10))
+   array([[ 0,  1,  2,  3,  4,  5,  6,  7,  8,  9],
+          [10, 11, 12, 13, 14, 15, 16, 17, 18, 19],
+          [20, 21, 22, 23, 24, 25, 26, 27, 28, 29],
+          [30, 31, 32, 33, 34, 35, 36, 37, 38, 39],
+          [40, 41, 42, 43, 44, 45, 46, 47, 48, 49],
+          [50, 51, 52, 53, 54, 55, 56, 57, 58, 59],
+          [60, 61, 62, 63, 64, 65, 66, 67, 68, 69],
+          [70, 71, 72, 73, 74, 75, 76, 77, 78, 79],
+          [80, 81, 82, 83, 84, 85, 86, 87, 88, 89],
+          [90, 91, 92, 93, 94, 95, 96, 97, 98, 99]])
+   >>> output_arr = kernel1(input_arr)
+   array([[  0.,   0.,   0.,   0.,   0.,   0.,   0.,   0.,   0.,   0.],
+          [  0.,  11.,  12.,  13.,  14.,  15.,  16.,  17.,  18.,   0.],
+          [  0.,  21.,  22.,  23.,  24.,  25.,  26.,  27.,  28.,   0.],
+          [  0.,  31.,  32.,  33.,  34.,  35.,  36.,  37.,  38.,   0.],
+          [  0.,  41.,  42.,  43.,  44.,  45.,  46.,  47.,  48.,   0.],
+          [  0.,  51.,  52.,  53.,  54.,  55.,  56.,  57.,  58.,   0.],
+          [  0.,  61.,  62.,  63.,  64.,  65.,  66.,  67.,  68.,   0.],
+          [  0.,  71.,  72.,  73.,  74.,  75.,  76.,  77.,  78.,   0.],
+          [  0.,  81.,  82.,  83.,  84.,  85.,  86.,  87.,  88.,   0.],
+          [  0.,   0.,   0.,   0.,   0.,   0.,   0.,   0.,   0.,   0.]])
+   >>> input_arr.dtype
+   dtype('int64')
+   >>> output_arr.dtype
+   dtype('float64')
+
+Note that the stencil decorator has determined that the output type
+of the specified stencil kernel is ``float`` and has thus created the
+output array as ``float`` while the input array is of type ``int``.
+
+Stencil Parameters
+==================
+
+Stencil kernel definitions may take any number of arguments with
+the following provisions.  The first parameter must be an array.
+The size and shape of the output array will be the same as the
+first parameter.  Additional parameters may either be scalars or
+arrays.  For array parameters, those arrays must be at least as large
+as the first array in each dimension.  Array indexing is relative for
+all such input array parameters.
+
+.. _stencil-kernel-shape-inference:
+
+Kernel shape inference and border handling
+==========================================
```

Comment:
Since it is called neighborhood in the API, should be called the same here.  Just implemented a fix for this.

Reply:
Should mention this in the developer docs I think.  Should the ability to query this attribute be mentioned here in the user docs?

Reply:
I think it would be useful for both. As a user, a quick check that the relative indices cover the expected area may be quite useful as I imagine they may be prone to typos. Thanks for adding this.

Reply:
Done.